[TAN-164] Add support for ics format to GET .../event/:id endpoint

adessy opened the pull request:
# Changelog
## Technical
- Add support for ics format to the `GET .../event/:id` endpoint.

adessy committed:
[TAN-104] Add support for ics format to GET .../event/:id endpoint

adessy committed:
[TAN-104] Refactor MultilocService#t signature

There is no need in passing the user to MultilocService#t as it only uses the user's locale and none of the other user attributes. Passing the user to MultilocService#t forces all callers to be aware of the user object even if they don't need it.

This commit removes the user argument from MultilocService#t and replaces it with the preferred locale.

kogre commented on back/app/services/multiloc_service.rb:
```diff
@@ -5,12 +5,12 @@ def initialize(app_configuration: nil)
-    user_locale = user&.locale || I18n.locale.to_s
-    result = ([user_locale] + locales + translations.keys).each do |locale|
+    preferred_locale ||= I18n.locale.to_s
+    result = ([preferred_locale] + locales + translations.keys).each do |locale|
```
Unrelated to your code change (which I like), but opportunistically: Reading this I'm wondering whether it wouldn't be worth it to refactor this so we don't call app_configuration in the case where we have a preferred_locale.

The repeated queries to app_configuration are a significant slowdown factor in our BE code, and this small change could maybe speed us up in many places?

adessy replied:
Good point! I'll rework that part.

> Sure you've covered all relevant cases of service.t?

I tried my best :)

kogre reviewed (approved):
Looks good to me, really no remarks. Sure you've covered all relevant cases of `service.t`?

Just have one opportunistic suggestion.

adessy committed:
[TAN-104] Optimize MultilocService#t

Avoid unnecessary AppConfiguration instantiation, preventing potential extra DB queries.

adessy committed:
Merge branch 'master' into TAN-104/event-improvements/add-to-calendar-ics
